src/features/unified-share-modal/EmailForm.js

```diff
@@ -205,14 +207,27 @@ class EmailForm extends React.Component<Props, State> {
         const { intl } = this.props;
         let contactsFieldError = '';
 
-        if (text && !this.isValidEmail(text)) {
+        if (text && !emailValidator(text)) {
             contactsFieldError = intl.formatMessage(commonMessages.invalidEmailError);
         }
         this.setState({ contactsFieldError });
     };
 
-    isValidEmail = (text: string): boolean => {
-        return emailValidator(text);
+    isValidContactPill = (contactPill: string | Contact): boolean => {
```

Comment:
Another side-effect of enabling `allowInvalidPills` is that the validator also gets called with pill objects (contacts in this case) instead of just strings.